platforms/hyperledger-indy/charts/indy-ledger-txn/templates/job.yaml

```diff
@@ -0,0 +1,62 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: "{{ $.Values.metadata.name }}"
+  namespace: "{{ $.Values.metadata.namespace }}"
+  labels:
+    app: "{{ $.Values.metadata.name }}"
+    app.kubernetes.io/name: "{{ $.Values.metadata.name }}"
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+spec:
+  serviceName: "{{ $.Values.metadata.name }}"
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: "{{ $.Values.metadata.name }}"
+        app.kubernetes.io/name: "{{ $.Values.metadata.name }}"
+        app.kubernetes.io/instance: {{ .Release.Name }} 
+    spec:
+      restartPolicy: OnFailure
+      imagePullSecrets:
+        - name: "{{ $.Values.image.pullSecret }}"
+      serviceAccountName: "{{ $.Values.vault.serviceAccountName }}"
+      containers:
+        - name:  "{{ $.Values.image.initContainer.name }}"
+          image: "{{ $.Values.image.initContainer.repository }}"
+          imagePullPolicy: Always
+          env:
+           - name: VAULT_ADDR
+             value: "{{ $.Values.vault.address }}"
+           - name: ADMIN_PATH
+             value: "{{ $.Values.organization.name.adminIdentity.path }}"
+           - name: ADMIN_NAME
+             value: "{{ $.Values.organization.name.adminIdentity.name }}"
+           - name: IDENTITY_NAME
+             value:  "{{ $.Values.organization.name.newIdentity.name }}"
+           - name: IDENTITY_PATH
+             value: "{{ $.Values.organization.name.newIdentity.path }}"
+           - name: IDENTITY_ROLE
+             value: "{{ $.Values.organization.name.newIdentity.name }}"
+           - name: POOL_GENESIS_PATH
+             value: "{{ $.Values.configmap.poolGenesis }}"
+          command:
+            - "/bin/bash"
+            - "-c"
+            - >
+              apt-get install curl -y;
+              validateVaultResponse () {
+              if echo ${2} | grep "errors"; then
+               echo "ERROR: unable to retrieve ${1}: ${2}"
+               exit 1
+               fi
+              }
+
+              KUBE_TOKEN=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token);
+
+              curl --request POST --data '{"jwt": "'"$KUBE_TOKEN"'", "role": "{{ $.Values.vault.role }}"}' {{ $.Values.vault.address }}/v1/auth/kubernetes/login | jq -j '.auth.client_token' > token;
+              VAULT_TOKEN=$(cat token);
+              validateVaultResponse 'vault login token' "${VAULT_TOKEN}";
```

Comment:
You don't have a vault cli here (I've created a bug for the indy-key-mgmt job as well).

Reply:
So de we need to install the vault CLI here, Also the curl request wont require the vault CLI.

Reply:
I think we've already gone through the story, Vault doesn't offer a lightweight CLI, so you have to use just curl

Reply:
Yes Ive used curl request here, So what changes are exactly required here?

Reply:
Have you tested the job? This piece 'vault login token' won't work, you don't have the vault CLI installed in the Docker image.

Reply:
In order to configure the vault to talk to cluster, we need to authenticate it with kubernetes Service Account Token which must be present in the vault as a pre requisite. So when that  SA token gets verified the vault gives the vault token to container which can be used for furthur communication (as done in fabric and corda)

Reply:
How did you test the Helm chart in this PR?